### PR TITLE
Remove Asylo runtime variant

### PR DESCRIPTION
**Pull request by daviddrysdale**
# Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [x] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

**Comment by daviddrysdale** [comment]
For #709

File: cloudbuild.yaml [hunk 1]
```diff
@@ -152,11 +133,12 @@ steps:
 # - https://pantheon.corp.google.com/storage/browser/artifacts.oak-ci.appspot.com/test/?project=oak-ci
 # - https://cloud.google.com/cloud-build/docs/configuring-builds/store-images-artifacts#storing_artifacts_in
 # TODO: Finalize artifact location.
-artifacts:
-  objects:
-    location: gs://artifacts.oak-ci.appspot.com/test
-    paths:
-      - ./liboak_enclave_unsigned.so
+# TODO(#709): get example Docker image working with dev server
+#artifacts:
```
Comment (tiziano88) [inline cloudbuild.yaml]: Perhaps we should keep uploading the compiled oak_runtime (dev) binary? Not sure it's particularly useful by itself though.

Comment (daviddrysdale) [reply cloudbuild.yaml]: I don't know anything about how the artifact upload stuff works, so it seemed easiest just to disable it for the moment; maybe @ipetr0v can explain and/or fix it for me?

File: scripts/build_example [hunk 2]
```diff
@@ -85,21 +85,20 @@ case "${language}" in
...
     echo "Docker images are only supported for Rust modules"
     exit 1
   fi
 
   docker_bazel_build_flags=("${bazel_build_flags[@]}")
-  # Docker images only support the Asylo server.
-  docker_bazel_build_flags+=(
-    "--config=enc-sim"
-  )
   bazel build "${docker_bazel_build_flags[@]}" "//examples/${EXAMPLE}/server:${EXAMPLE}_image.tar"
```
[the remark this answers is not in the document]
Comment (daviddrysdale) [reply scripts/build_example]: Done, good spot.

**Pull request merged**